src/unix/aix-common.c

```diff
@@ -253,15 +266,16 @@ int uv_interface_addresses(uv_interface_address_t** addresses, int* count) {
     sa_addr = (struct sockaddr_dl*) &p->ifr_addr;
     memcpy(address->phys_addr, LLADDR(sa_addr), sizeof(address->phys_addr));
 
-    if (ioctl(sockfd, SIOCGIFNETMASK, p) == -1) {
-      uv__close(sockfd);
-      return UV_ENOSYS;
```

Comment:
I guess it's unrelated to this PR, but it looks like this function is not properly cleaning up after itself if an error occurs within the loop (`*addresses` is not freed, any memory allocated by `uv__strdup()` is not freed).

Reply:
I'll see what I can do.

Reply:
Pushed a commit to do cleanup on the error paths.